### PR TITLE
fix: pytest imports for isolation forests

### PR DESCRIPTION
**Pull request by thatlittleboy**
## Overview

Description of the changes proposed in this pull request:

* The pytest importorskip statements were written incorrectly, resulting in the IsolationForest tests being skipped all the time. This PR fixes that.

## Checklist

- Closes #xxxx  
- [X] All [pre-commit checks](https://pre-commit.com/#install) pass.
- Unit tests added (if fixing a bug or adding a new feature)
- Added entry to `CHANGELOG.md` (if changes will affect users)

### CODE REVIEW

File: tests/explainers/test_tree.py [hunk 1]
```diff
@@ -387,8 +387,9 @@ def test_catboost_categorical():
 
 
 def test_isolation_forest():
-    IsolationForest = pytest.importorskip("sklearn.ensemble.IsolationForest")
-    _average_path_length = pytest.importorskip("sklearn.ensemble._iforest._average_path_length")
+    from sklearn.ensemble import IsolationForest
+    from sklearn.ensemble._iforest import _average_path_length
```
Comment (connortann) [inline tests/explainers/test_tree.py]: We should be cautious about importing private objects from other libraries: `_iforest` is not part of sklearn's public API and could change or break without notice.

Would it be better to copy the relevant [private function](https://github.com/scikit-learn/scikit-learn/blob/150ff34e85f91a3a19a29dc4e19d50131fbb0251/sklearn/ensemble/_iforest.py#L531) and vendor it here?

**Review by connortann — CHANGES_REQUESTED** [review]

File: tests/explainers/test_tree.py (hunk 1 above)
Comment (thatlittleboy) [reply tests/explainers/test_tree.py]: done

File: tests/explainers/test_tree.py [hunk 2]
```diff
@@ -386,9 +386,37 @@ def test_catboost_categorical():
 # TODO: Test tree_limit argument
 
 
+def _average_path_length(n_samples_leaf):
+    """
+    Vendored from: https://github.com/scikit-learn/scikit-learn/blob/399131c8545cd525724e4bacf553416c512ac82c/sklearn/ensemble/_iforest.py#L531
+
+    For use in isolation forest tests.
+    """
+    from sklearn.utils import check_array
```
Comment (connortann) [inline tests/explainers/test_tree.py]: Nit: as a core dependency, this import could follow PEP8 and go at the top.

**Review by connortann — APPROVED** [review]
One optional nitpick but otherwise LGTM!

**Pull request merged**